Core/include/Acts/EventData/detail/ParticleHypothesis.hpp

```diff
@@ -0,0 +1,113 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2023 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Units.hpp"
+#include "Acts/EventData/detail/Charge.hpp"
+#include "Acts/Utilities/PdgParticle.hpp"
+
+#include <utility>
+
+namespace Acts {
+namespace detail {
+
+/// @brief Particle hypothesis used in reconstruction
+///
+/// Our reconstruction hypothesis consists of absolute PDG code, mass and
+/// absolute charge
+template <typename charge_t>
+class ParticleHypothesis {
+ public:
+  using ChargeType = charge_t;
+
+  constexpr static ParticleHypothesis<ChargeType> muon() {
+    return ParticleHypothesis<ChargeType>(
+        PdgParticle::eMuon, 105.6583755 * UnitConstants::MeV, UnitConstants::e);
```

Comment:
Should we expose these constants also in a header file under `Core/Definitions`?

Reply:
I also wondered that but then I thought this header is actually doing exactly that. But I guess it is kinda hidden in `EventData/detail`